adapters/repos/db/search.go

```diff
@@ -326,6 +348,9 @@ func (db *DB) ResolveReferences(ctx context.Context, objs search.Results,
 	props search.SelectProperties, groupBy *searchparams.GroupBy,
 	addl additional.Properties, tenant string,
 ) (search.Results, error) {
+	if global.Manager().RejectRequests() {
+		return nil, fmt.Errorf("shutdown in progress")
```

Comment:
Should the error message say `"server is shutting down"` like it does [here](https://github.com/weaviate/weaviate/pull/7998/files#diff-6f454b28f1d7b2f00dc49220806d8b0ba25c155c06d3a145c6e339ed3dab926aR39)?

Are the two messages interchangeable? IIUC, the difference between `RejectRequests` and `IsShutdownInProgress` is mainly their "target audience" -- IsShutdownInProgress is meant to interrupt some internal jobs (async), whereas RejectRequests is to be checked by the API server (sync). Plus the 5s delay.

Should there be a `global.ErrServerShuttingDown` to support error unwrapping and such?

Reply:
i'd say lets have global error variable to be used to be consistent